src/commons/authentication/ldap.rs

```diff
@@ -49,6 +55,22 @@ pub struct AuthenticationProvider {
 }
 
 impl AuthenticationProvider {
+    /// Returns the LDAP endpoint [`Url`].
+    pub fn endpoint_url(&self) -> Result<Url> {
+        let url = Url::parse(&format!(
+            "{protocol}{server_hostname}:{server_port}",
+            protocol = match self.tls.tls {
+                None => "ldap://",
+                Some(_) => "ldaps://",
+            },
+            server_hostname = self.hostname,
+            server_port = self.port()
+        ))
```

Comment:
Nit: Is there any particular reason we first assign `self.*` to a formatting variable. I don't know if there is any difference performance-wise, but we might want to think about a rule to add to our style guide anyways...

Reply:
Do I understand you correctly that you want anonymous formatting arguments?
I would have expected no performance difference and would slightly prefer named arguments for readability

Reply:
I would expect no performance difference either, but you never know. I'm happy leaving it as is (and yes, named arguments are better for readability), but I then would prefer that we write this down in our code style guide :)

Reply:
I will create an Arch Meeting item for this. I might also add a PR in our https://github.com/stackabletech/documentation repo to have the code style guide visible in our public documentation.

Reply:
Agreed :+1:

Reply:
I'm not sure I understood what you meant, but it sounded like you are talking about inlining?

You can't (AFAIK) do `format!("{self.hostname}")` for example.

Did you mean to declare variables and then inline them?

```rust
let protocol = match ... {};
let server_hostname = self.hostname;
let server_port = self.port;

let url = Url::parse(&format!("{protocol}{server_hostname}:{server_port}"))
// ...
```

Reply:
Ah I see. It's just about named vs unnamed. I think named is best, so that order doesn't matter and there is less chance of accidents.

But I also prefer inlining when possible (no extra args).

Reply:
Just had a look and commented on your Arch meeting entry @Techassi